File: RB101_Programming_Foundations/small_problems/Easy1/Run2/stringy_strings.rb
def stringy(number, start=1)
  base = start == 1 ? '10' : '01'
  result = base * (number / 2)
  number.odd? ? result + base[0] : result
end


puts stringy(6) == '101010'
puts stringy(9) == '101010101'
puts stringy(4) == '1010'
puts stringy(7) == '1010101'

File: js210_fundamentals_of_JS_Programming/problem_sets/easy2/stringy_strings.js
/* Algorithms
create a string that store the result
create a [1, 0] array
iterate from index 0 to n - 1
  check if the current index is odd or even
  add the respective character to the result string
return the string
*/

function stringy(length) {
  let result = '';
  let numbers = ['1', '0'];

  for (let index = 0; index < length; index++) {
    result += numbers[index % 2];
  }
  return result;
}

console.log(stringy(6)); // "101010"
console.log(stringy(9)); // "101010101"
console.log(stringy(4)); // "1010"
console.log(stringy(7)); // "1010101"
